lib/mqtt_abstraction_impl.cpp

```diff
@@ -306,7 +306,7 @@ void MQTTAbstractionImpl::on_mqtt_message(std::shared_ptr<Message> message) {
 
             if (topic_matches) {
                 found = true;
-                handler.add({topic, data});
+                handler.add(std::unique_ptr<ParsedMessage>(new ParsedMessage{topic, std::move(data)}));
```

Comment:
@a-w50 
This line can be called multiple times. It is within:
`for (auto& [handler_topic, handler] : this->message_handlers)`.

The first time it is called `data` is moved to the handler.
The second and subsequent calls `data` is empty.